### PR TITLE
websocket clean up resolvers after client abruptly disconnects

### PR DESCRIPTION
**Pull request by RobertoOrtis**
Describe your PR and link to any relevant issues. 
Extends https://github.com/99designs/gqlgen/pull/1248 https://github.com/99designs/gqlgen/pull/1248#issuecomment-897376487

Websocket resolvers were not being cleaned up after the client abruptly disconnects. For example, when the users close or refresh their browsers. This fixes it by cleaning active resolvers whenever a `sendConnectionError` is called.

I have:
 - [ ] Added tests covering the bug / feature (see [testing](https://github.com/99designs/gqlgen/blob/master/TESTING.md))
 - [ ] Updated any relevant documentation (see [docs](https://github.com/99designs/gqlgen/tree/master/docs/content))

### CODE REVIEW

**Comment by StevenACoffman** [comment]
@zdraganov This is another PR that seems to overlap a little with some of the other more recently merged websock PRs. Do you think there's anything of value here that can be freshened up?

**Comment by zdraganov** [comment]
I will take a look.

File: graphql/handler/transport/websocket.go [hunk 1]
```diff
@@ -283,12 +285,23 @@ func (c *wsConnection) sendError(id string, errors ...*gqlerror.Error) {
...
+	for i, _ := range c.active {
+		select {
+		case <- c.ctx.Done():
+			b, err := json.Marshal(&gqlerror.Error{Message: fmt.Sprintf(format, args...)})
+			if err != nil {
+				panic(err)
+			}
 
-	c.write(&operationMessage{Type: connectionErrorMsg, Payload: b})
+			c.write(&operationMessage{Type: connectionErrorMsg, Payload: b})
```
[the remark this answers is not in the document]
Comment (zdraganov) [reply graphql/handler/transport/websocket.go:296]: Do we need to write same error in loop here?

File: graphql/handler/transport/websocket.go [hunk 2]
```diff
@@ -127,9 +127,11 @@ func (c *wsConnection) init() bool {
 }
 
 func (c *wsConnection) write(msg *operationMessage) {
-	c.mu.Lock()
-	c.conn.WriteJSON(msg)
-	c.mu.Unlock()
+	if msg.Type == "data" {
```
[the remark this answers is not in the document]
Comment (zdraganov) [reply graphql/handler/transport/websocket.go:130]: Yeah @jaredly is right. We cannot afford to miss other types of messages here.

**Comment by zdraganov** [comment]
I think this is more related to this one https://github.com/99designs/gqlgen/pull/1728. I think this can be closed as it's a duplicate. @RobertoOrtis do you think there's something more on top of the mentioned PR?

**Comment by StevenACoffman** [comment]
I appreciate your contribution, but I am closing this as a duplicate. If there's something else I missed, please don't hesitate to open a new PR.

**Pull request closed without merging**